types/views/views.go

```diff
@@ -181,3 +193,152 @@ func (v IPPrefixSlice) MarshalJSON() ([]byte, error) {
 func (v *IPPrefixSlice) UnmarshalJSON(b []byte) error {
 	return v.ж.UnmarshalJSON(b)
 }
+
+// containsPointers reports whether the provided type is a pointer or wraps any
+// pointers.
+func containsPointers(t reflect.Type) bool {
```

Comment:
(I thought I made this comment, but I can't find any record of it. Strange.)

I think this is too narrow. What we really want is to know whether this contains any mutable values.

In Go, this can handle through the following kinds:
* pointers
* slices (since the user can overwrite elements or take the address of them)
* maps (since the user can overwrite or delete an entry value)

We may also want to reject the following kinds:
* channels
* funcs

since those are technically mutable (but there is also no obvious immutable view of them).